### PR TITLE
combine linux and win Dockerfiles using build targets

### PR DESCRIPTION
**Pull request by rbtr**
**Reason for Change**:

collapses the $OS.Dockerfiles in to a single Dockerfile and combines build stages where possible. This helps us maintain consistency between the containers built for different OSes.
Updated for CNS, azure-cni, and azure-ipam.
NPM Dockerfiles were not updated except to add target labels to the final image so that they would continue to build with the updated Make recipe.

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] relevant PR labels added

**Notes**:

### CODE REVIEW

**Review by huntergregory — APPROVED** [review]

File: cni/Dockerfile [hunk 1]
```diff
@@ -11,14 +11,15 @@ COPY . .
 RUN GOOS=$OS CGO_ENABLED=0 go build -a -o /go/bin/azure-vnet -trimpath -ldflags "-X main.version="$VERSION"" -gcflags="-dwarflocationlists=true" cni/network/plugin/main.go
 RUN GOOS=$OS CGO_ENABLED=0 go build -a -o /go/bin/azure-vnet-telemetry -trimpath -ldflags "-X main.version="$VERSION"" -gcflags="-dwarflocationlists=true" cni/telemetry/service/telemetrymain.go
 RUN GOOS=$OS CGO_ENABLED=0 go build -a -o /go/bin/azure-vnet-ipam -trimpath -ldflags "-X main.version="$VERSION"" -gcflags="-dwarflocationlists=true" cni/ipam/plugin/main.go
-RUN GOOS=$OS CGO_ENABLED=0 go build -a -o /go/bin/azurecni-stateless -trimpath -ldflags "-X main.version="$VERSION"" -gcflags="-dwarflocationlists=true" cni/network/stateless/main.go
+RUN GOOS=$OS CGO_ENABLED=0 go build -a -o /go/bin/azure-vnet-stateless -trimpath -ldflags "-X main.version="$VERSION"" -gcflags="-dwarflocationlists=true" cni/network/stateless/main.go
```
Comment (jpayne3506) [inline cni/Dockerfile:14]: Does this impact anything?

Comment (rbtr) [reply cni/Dockerfile:14]: no, not until k8s 1.30 at least
cc @behzad-mir for awareness though

**Review by jpayne3506 — APPROVED** [review]

**Review by thatmattlong — APPROVED** [review]

**Comment by pjohnst5** [comment]
👀

**Review by matmerr — DISMISSED** [review]
very pro build targets after using them elsewhere

**Review by pjohnst5 — APPROVED** [review]

**Review by jpayne3506 — APPROVED** [review]

**Review by thatmattlong — APPROVED** [review]

**Review by jpayne3506 — DISMISSED** [review]
🟢 Pending Pipelines

**Review by jpayne3506 — APPROVED** [review]

**Review by thatmattlong — APPROVED** [review]

**Review by huntergregory — APPROVED** [review]
approving from NPM perspective 👍

**Pull request merged**